source/base/quadrature.cc

```diff
@@ -59,6 +59,31 @@ Quadrature<dim>::initialize(const std::vector<Point<dim>> &p,
 
 
 
+template <int dim>
+void
+Quadrature<dim>::initialize(const ArrayView<const Point<dim>> &points,
+                            const ArrayView<const double>     &weights)
+{
+  this->weights.clear();
+  if (!weights.empty())
```

Comment:
Like in https://github.com/dealii/dealii/pull/15661#pullrequestreview-1515483214, I find `!empty()` hard to read because it's a double negation. Would you mind changing that to `size() > 0`?